doc/guidelines/e2e_frontend_coding_guidelines.md

```diff
@@ -0,0 +1,145 @@
+# End-to-End Testing Guidelines
+
+## Frontend Coding Guidelines
+
+These guidelines will help you write frontend code that enables resilient or user centric test execution.
+Currently these guidelines are strongly oriented according to Playwrights Best Practices.
+When following the below mentioned four points, you will be on a good way to write good testable code.
+
+### 1. Locators
+
+In order to find elements on the webpage the tests will need to use one of the following [locators](https://playwright.dev/docs/locators):
+
+- getByRole() to locate by explicit and implicit accessibility attributes.
+- getByText() to locate by text content.
+- getByLabel() to locate a form control by associated label's text.
+- getByPlaceholder() to locate an input by placeholder.
+- getByAltText() to locate an element, usually image, by its text alternative.
+- getByTitle() to locate an element by its title attribute.
+- getByTestId() to locate an element based on its `data-testid` attribute (other attributes can be configured).
```

Comment:
What means "other attributes can be configured"? Let us keep it as simple as possible, the list above looks like exhaustive and not minimal?

Reply:
You can change the attribute name to something else if you dont like `data-testid`.